segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -268,14 +268,11 @@ public QueueStats getQueueStatistics() {
     private void processWritesSync() {
         if (getWriteLedger().ledger.isClosed()) {
             // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
-            // the write processor upon finish, so
+            // the write processor upon finish, so the writes can be reattempted.
             this.rolloverProcessor.runAsync();
-        } else if (!processPendingWrites()) {
+        } else if (!processPendingWrites() && !this.closed.get()) {
             // We were not able to complete execution of all writes. Try again.
             this.writeProcessor.runAsync();
-        } else {
```

Comment:
Well ... the description lists that under how it used to work before, and why it was flawed. 
The master bullet says `This was due to the fact that:`.